src/main/java/io/logz/jmx2logzio/clients/JavaAgentClient.java

```diff
@@ -55,28 +58,38 @@ public JavaAgentClient() {
      * @throws MBeanClientPollingFailure
      */
     @Override
-    public List<MetricBean> getBeans() throws MBeanClientPollingFailure {
+    public List<MetricBean> getBeans() {
 
-        try {
             List<MetricBean> metricBeans = Lists.newArrayList();
             Set<ObjectInstance> instances = server.queryMBeans(null, null);
 
             for (ObjectInstance instance : instances) {
-                MBeanInfo mBeanInfo = server.getMBeanInfo(instance.getObjectName());
-                List<String> attributes = Lists.newArrayList();
-
-                for (MBeanAttributeInfo attribute : mBeanInfo.getAttributes()) {
-                    attributes.add(attribute.getName());
+                instancesCount++;
+                try {
+                    MBeanInfo mBeanInfo = server.getMBeanInfo(instance.getObjectName());
+                    List<String> attributes = Lists.newArrayList();
+
+                    for (MBeanAttributeInfo attribute : mBeanInfo.getAttributes()) {
+                        attributes.add(attribute.getName());
+                    }
+
+                    // Dont change to getCanonicalName(), we need it to preserve the order so we can have a valuable metrics tree
+                    metricBeans.add(new MetricBean(instance.getObjectName().getDomain() + ":" + instance.getObjectName().getKeyPropertyListString(), attributes));
+                } catch (InstanceNotFoundException e) {
+                    logger.debug("Instance Not found: " + e.getMessage(), e);
+                    instanceNotFoundCount++;
+                }  catch (IntrospectionException | ReflectionException e) {
+                    logger.warn(e.getMessage(), e);
                 }
-
-                // Dont change to getCanonicalName(), we need it to preserve the order so we can have a valuable metrics tree
-                metricBeans.add(new MetricBean(instance.getObjectName().getDomain() + ":" + instance.getObjectName().getKeyPropertyListString(), attributes));
+            }
+            if (instancesCount >= 100 ) {
+                if (((double)instanceNotFoundCount / (double)instancesCount) * 100 > INSTANCES_NOT_FOUND_PERCENTAGE_WARNING_THRESHOLD) {
```

Comment:
If I'm not mistake the first `(double)` is enough. Java makes all other expressions double as well.

Reply:
You are right.